criu/include/sched.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __CR_SCHED_H__
+#define __CR_SCHED_H__
+
+#include <linux/types.h>
+
+#ifndef ptr_to_u64
+#define ptr_to_u64(ptr)	((__u64)((uintptr_t)(ptr)))
+#endif
+#ifndef u64_to_ptr
+#define u64_to_ptr(x)	((void *)(uintptr_t)x)
+#endif
+
+#ifndef CLONE_ARGS_SIZE_VER0
+#define CLONE_ARGS_SIZE_VER0
+
+/* This structure is needed by clone3(). */
+
+struct _clone_args {
```

Comment:
And why is this called `_clone_args` and not `_clone_args`? :)

Reply:
```c
#ifndef CLONE_ARGS_SIZE_VER0
#define CLONE_ARGS_SIZE_VER0 64
struct clone_args {
	__aligned_u64 flags;
	__aligned_u64 pidfd;
	__aligned_u64 child_tid;
	__aligned_u64 parent_tid;
	__aligned_u64 exit_signal;
	__aligned_u64 stack;
	__aligned_u64 stack_size;
	__aligned_u64 tls;
#define CLONE_ARGS_SIZE_VER1 80
	__aligned_u64 set_tid;
	__aligned_u64 set_tid_size;
};
#endif /* CLONE_ARGS_SIZE_VER0 */
```
I think the ifdef for CLONE_ARGS_SIZE_VER1 is not needed. A system where `VER0` is defined but not `VER1` is broken.